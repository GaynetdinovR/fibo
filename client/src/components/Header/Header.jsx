import styles from "../../styles/components/Header.module.sass";
import HeaderNavBar from "./components/HeaderNavBar.jsx";
import HeaderTopSide from "./components/HeaderTopSide.jsx";
import { useState } from "react";

const Header = () => {
	const [isMenuOpen, setMenu] = useState(false);

	return (
		<header className={styles.header}>
			<HeaderTopSide isMenuOpen={isMenuOpen} setMenu={setMenu}/>
			<HeaderNavBar isMenuOpen={isMenuOpen}/>
		</header>
	);
};

export default Header;