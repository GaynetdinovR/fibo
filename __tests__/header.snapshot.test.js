import renderer from "react-test-renderer";
import { it, expect } from "@jest/globals";
import Header from "../src/components/Header/Header.jsx";

it("renders correctly", () => {
	const tree = renderer.create(<Header />).toJSON();
	expect(tree).toMatchSnapshot();
});
