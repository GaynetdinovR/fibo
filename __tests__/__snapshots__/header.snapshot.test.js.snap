// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="header__top_side"
  >
    <a
      className="header__logo"
    >
      <img
        alt="logo"
        src="./icons/logo_1.png"
      />
    </a>
    <div
      className="header__left_top_side"
    >
      <div
        className="header__location"
      >
        <span>
           Доставка пасты 
        </span>
        <span
          className="header__city"
        >
           Москва 
        </span>
      </div>
      <div
        className="header__service"
      >
        <div
          className="header__service_img"
        >
          <img
            alt="yandex_food"
            src="./icons/yandex_food.png"
          />
        </div>
        <span
          className="header__service_name"
        >
          Яндекс Еда
        </span>
        <div
          className="header__service_grade"
        >
          <span>
            4.8
          </span>
          <div>
            <img
              alt="star"
              src="./icons/star.png"
            />
          </div>
        </div>
        <div
          className="header__delivery_time"
        >
          <span>
            Время доставки
          </span>
          <span>
            от 31 мин
          </span>
        </div>
      </div>
    </div>
    <div
      className="header__right_top_side"
    >
      <button
        className="header__get_call"
      >
        Заказать звонок
      </button>
      <a
        className="header__phone_number"
        href="tel:84993918449"
      >
        8 499 391-84-49
      </a>
    </div>
  </div>
  <nav
    className="header__navbar"
  >
    <ul
      className="header__list"
    >
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Пицца
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Паста
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Супы
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Салаты
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Напитки
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Десерты
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Бакалея
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Антипасти
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Акции
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Комбо
        </a>
      </li>
      <li
        className="header__item"
      >
        <a
          className="header__link"
        >
          Контакты
        </a>
      </li>
    </ul>
    <button
      className="header__auth"
    >
      Войти
    </button>
    <button
      className="header__cart_btn default_button"
    >
      Корзина | 0
    </button>
  </nav>
</header>
`;
